interlok-core/src/main/java/com/adaptris/core/services/splitter/PooledSplitJoinService.java

```diff
@@ -0,0 +1,437 @@
+ * service for handling split/aggregation since it has more predictable peformance characteristics
+ * in constrained environments where the numbers of messages that are generated and aggregated can
+ * be large/unknown.
+ * </p>
+ * <p>
+ * This service splits a message according to the configured {@link MessageSplitter} implementation,
+ * executes the configured {@link Service} and subsequently aggregates all the messages back using
+ * the configured {@link MessageAggregator} implementation
+ * <p>
+ * <p>
+ * A pool of {@link com.adaptris.core.Service} instances is maintained and re-used for each message;
+ * the cost of initialisation for the wrapped service, is incurred during this service's
+ * initialisation phase.
+ * </p>
+ * <p>
+ * Aggregation may start happening as soon as messages are available to be aggregated (i.e. a split
+ * message has been operated on) using the
+ * {@link MessageAggregator#aggregate(AdaptrisMessage, Iterable)} method. Performance characterstics
+ * will largely depend on how the splitter and aggregator implementations iterate over the messages.
+ * </p>
+ *
+ * @config pooled-split-join-service
+ */
+@XStreamAlias("pooled-split-join-service")
+@AdapterComponent
+@ComponentProfile(
+    summary = "Split a message and then execute the associated services on the split items, aggregating the split messages afterwards",
+    since = "3.11.1", tag = "service,splitjoin")
+@DisplayOrder(order =
+{
+    "splitter", "service", "aggregator", "timeout", "poolsize", "sendEvents",
+    "serviceErrorHandler"
+})
+@NoArgsConstructor
+public class PooledSplitJoinService extends ServiceImp implements EventHandlerAware, ServiceWrapper {
+
+  private static final long DEFAULT_TTL = TimeUnit.MINUTES.toMillis(10L);
+  private static final int DEFAULT_POOLSIZE = 10;
+  private static final long UNSET_SPLIT_COUNT = -1;
+
+  /**
+   * The {@link com.adaptris.core.Service} to execute over all the split messages.
+   *
+   */
+  @NotNull
+  @Valid
+  @Getter
+  @Setter
+  @NonNull
+  private Service service;
+
+  /**
+   * The {@link MessageSplitter} implementation to use to split the incoming message.
+   *
+   */
+  @NotNull
+  @Valid
+  @Getter
+  @Setter
+  @NonNull
+  private MessageSplitter splitter;
+  /**
+   * The {@link MessageAggregator} implementation to use to join messages together.
+   *
+   */
+  @NotNull
+  @Valid
+  @Getter
+  @Setter
+  @NonNull
+  private MessageAggregator aggregator;
+
+  /**
+   * The max amount of time to wait for all the operations to complete.
+   * <p>
+   * If not explicitly specified then is set to be 10 minutes; in the event that the timeout is
+   * exceeded, then an exception will be thrown eventually.
+   * </p>
+   */
+  @Valid
+  @Getter
+  @Setter
+  private TimeInterval timeout;
+  @InputFieldDefault(value = "false")
+  @Getter
+  @Setter
+  @AdvancedConfig
+  private Boolean sendEvents;
```

Comment:
Javadoc is missing. Maybe we can use the same as in AdvancedMessageSplitterService#setSendEvents